src/Microsoft.Kusto.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -671,6 +671,28 @@ internal string SelectFromTableOrView(IDataSource dataSource, Urn urn)
 
             return selectQuery.ToString();
         }
+        
+        internal string AlterFunction(IDataSource dataSource, ScriptingObject scriptingObject)
+        {
+            var functionName = scriptingObject.Name.Substring(0, scriptingObject.Name.IndexOf('('));
+            
+            var kustoDataSource = dataSource as KustoDataSource;
```

Comment:
Let's not leak the specific implementation here. We should be talking to IDataSource to get the function info. Not to KustoDataSource. ScripterCore should be unaware of which particular data source it is talking to.

Reply:
Great point!